.travis.yml

```diff
@@ -1,32 +1,69 @@
-  - sudo apt-get update
-  - sudo apt-get install -y opam aspcud time libgtk2.0-dev libgtksourceview2.0-dev emacs
-# build coqide along with the Tactics package, just because it's a short package
+language: generic
+sudo: false
+
+services:
+  - docker
+
 env:
-  - PACKAGES="Foundations Combinatorics Algebra NumberSystems PAdics"
-  - PACKAGES=CategoryTheory
-  - PACKAGES="MoreFoundations Ktheory"
-  - PACKAGES=HomologicalAlgebra
-  - PACKAGES=Topology
-  - PACKAGES=RealNumbers
-  - PACKAGES=SubstitutionSystems
-  - PACKAGES=Tactics BUILD_COQIDE=yes BUILD_ALSO=TAGS
-  - PACKAGES=Folds
-  - FOUNDATIONS_CHANGE_ERROR=yes BUILD_ALSO="check-for-change-to-Foundations enforce-listing-of-proof-files"
-# building Coq in a separate stage folds up the output in the log:
+  global:
+    - THIS_REPO=UniMath
+  matrix:
+    - PACKAGES="Foundations Combinatorics Algebra NumberSystems PAdics"
+    - PACKAGES=CategoryTheory
+    - PACKAGES="MoreFoundations Ktheory"
+    - PACKAGES=HomologicalAlgebra
+    - PACKAGES=Topology
+    - PACKAGES=RealNumbers
+    - PACKAGES=SubstitutionSystems
+    - PACKAGES=Tactics BUILD_COQIDE=yes BUILD_ALSO=TAGS
+    - PACKAGES=Folds
+    - FOUNDATIONS_CHANGE_ERROR=yes BUILD_ALSO="check-for-change-to-Foundations enforce-listing-of-proof-files"
+
+# The "docker run" command will pull if needed.
+# Running this first gives two tries in case of network lossage.
 before_script:
-  - opam init --yes --no-setup
-  - eval $(opam config env)
-  - opam install lablgtk camlp5 num --yes --verbose
-  - time make build-coq
+  - timeout 5m docker pull palmskog/xenial-unimath || true
+
+# Using travis_wait here seems to cause the job to terminate after 1 minute
+# with no error (!).
+# The git commands are tried twice, in case of temporary network failure.
+# The fcntl line works around a bug where Travis truncates logs and fails.
 script:
-  - time make TIMECMD=time $PACKAGES $BUILD_ALSO FOUNDATIONS_CHANGE_ERROR=$FOUNDATIONS_CHANGE_ERROR
+  - python -c "import fcntl; fcntl.fcntl(1, fcntl.F_SETFL, 0)"
+  - REMOTE_ORIGIN_URL=`git config --get remote.origin.url`
+  - echo "THIS_REPO=${THIS_REPO}"
+  - echo "PACKAGES=${PACKAGES}"
+  - echo "BUILD_COQIDE=${BUILD_COQIDE}"
+  - echo "BUILD_ALSO=${BUILD_ALSO}"
+  - echo "FOUNDATIONS_CHANGE_ERROR=${FOUNDATIONS_CHANGE_ERROR}"
+  - echo "TRAVIS_BRANCH=${TRAVIS_BRANCH}"
+  - echo "REMOTE_ORIGIN_URL=${REMOTE_ORIGIN_URL}"
+  - echo "TRAVIS_EVENT_TYPE=${TRAVIS_EVENT_TYPE}"
+  - echo "TRAVIS_COMMIT=${TRAVIS_COMMIT}"
+  - echo "TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST}"
+  - echo "TRAVIS_PULL_REQUEST_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH}"
+  - echo "TRAVIS_PULL_REQUEST_SHA=${TRAVIS_PULL_REQUEST_SHA}"
+  - echo "TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}"
+  - >-
+    docker run palmskog/xenial-unimath /bin/bash -c "true &&
+       if [ $TRAVIS_EVENT_TYPE = pull_request ] ; then
+         git clone --quiet --depth 9 $REMOTE_ORIGIN_URL $THIS_REPO || git clone --quiet --depth 9 $REMOTE_ORIGIN_URL $THIS_REPO
```

Comment:
It would be good to precede this section with a "set -x", so we can see the commands being executed.
Similarly for the else clause.

Reply:
I'll pass the `-x` option to bash itself instead and see what happens. I'm not sure this will make Travis output more readable though.